### PR TITLE
Add CodeQL.yml in Pull Request process to enable static analysis

### PR DESCRIPTION
**Pull request by goagain**
Enable CodeQL to enable static analysis.
The config `codeql.yml` is auto generated by GitHub. It enables static analysis for C# and Python. (It doesn't support bash and shell)

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (kyle-rader) [inline .github/workflows/codeql.yml:1]: @goagain Do we need this stand alone action if it's already being added to the release pipeline?

**Review by kyle-rader — CHANGES_REQUESTED** [review]
Not sure that we need this separate workflow.

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (goagain) [reply .github/workflows/codeql.yml:1]: I am going to move CodeQL to build and test yml.

**Review by kyle-rader — CHANGES_REQUESTED** [review]
@goagain This looks good, but need to merge with main from the CI updates I merged yesterday, sorry!

**Review by kyle-rader — CHANGES_REQUESTED** [review]
Lines 23-25 were removed in a prior PR, perhaps they made it back while resolving conflicts? They can be removed.

**Review by kyle-rader-msft — APPROVED** [review]

**Pull request merged**